lib/manager/poetry/readme.md

```diff
@@ -0,0 +1,170 @@
+## Overview
+
+#### Name of package manager
+
+[Poetry](https://poetry.eustace.io/)
+
+---
+
+#### What language does this support?
+
+Python
+
+---
+
+#### Does that language have other (competing?) package managers?
+
+pip
+
+## Package File Detection
+
+#### What type of package files and names does it use?
+
+_pyproject.toml_ and _poetry.lock_
+
+---
+
+#### What [fileMatch](https://renovatebot.com/docs/configuration-options/#filematch) pattern(s) should be used?
+
+_pyproject.toml_ and _poetry.lock_
+
+---
+
+#### Is it likely that many users would need to extend this pattern for custom file names?
+
+No
+
+---
+
+#### Is the fileMatch pattern likely to get many "false hits" for files that have nothing to do with package management?
+
+No
+
+## Parsing and Extraction
+
+#### Can package files have "local" links to each other that need to be resolved?
+
+Yes
+
+#### Is there reason why package files need to be parsed together (in serial) instead of independently?
+
+No
+
+---
+
+#### What format/syntax is the package file in? e.g. JSON, TOML, custom?
+
+TOML
+
+---
+
+#### How do you suggest parsing the file? Using an off-the-shelf parser, using regex, or can it be custom-parsed line by line?
+
+TOML can be parsed line by line
+
+---
+
+#### Does the package file structure distinguish between different "types" of dependencies? e.g. production dependencies, dev dependencies, etc?
+
+No
```

Comment:
I don't thinks that's correct
There are dependencies, dev-dependencies, and optional dependencies
See https://github.com/sdispater/poetry#introduction

```
[tool.poetry.dependencies]
python = "~2.7 || ^3.2"  # Compatible python versions must be declared here
toml = "^0.9"
# Dependencies with extras
requests = { version = "^2.13", extras = [ "security" ] }
# Python specific dependencies with prereleases allowed
pathlib2 = { version = "^2.2", python = "~2.7", allows-prereleases = true }
# Git dependencies
cleo = { git = "https://github.com/sdispater/cleo.git", branch = "master" }

# Optional dependencies (extras)
pendulum = { version = "^1.4", optional = true }

[tool.poetry.dev-dependencies]
pytest = "^3.0"
```

Reply:
Perhaps I misinterpreted the intent.
I read this has asking if the file system does anything different between `dependencies` and `dev-dependencies`.
Which it does not, it only labels them for developer convenience in the package file.

@rarkins clarification?

Reply:
@ChristianMurphy well, not only
You can do `poetry install --no-dev` to avoid installing dev-dependencies, can be especially useful inside Dockerfile, for CI and building lib for example